dicom_util/src/main/java/com/google/cloud/healthcare/DicomWebClient.java

```diff
@@ -128,18 +147,73 @@ public void stowRs(InputStream in) throws IDicomWebClient.DicomWebException {
     } catch (HttpResponseException e) {
       throw new DicomWebException(
           String.format("StowRs: %d, %s", e.getStatusCode(), e.getStatusMessage()),
-          e, e.getStatusCode(), Status.ProcessingFailure);
+          e,
+          e.getStatusCode(),
+          Status.ProcessingFailure);
     } catch (IOException e) {
       throw new IDicomWebClient.DicomWebException(e);
-    }
-    finally {
+    } finally {
       try {
         if ((resp) != null) {
           resp.disconnect();
         }
-      } catch(IOException e) {
+      } catch (IOException e) {
         throw new IDicomWebClient.DicomWebException(e);
       }
     }
   }
+
+  /** Retry the STOW-RS on an HTTP409, after DELETE. */
+  @Override
+  public Boolean getStowOverwrite() {
+    return this.useStowOverwrite;
+  }
+
+  /**
+   * Deletes the resource and returns void.
+   *
+   * @param path The resource URL path to delete.
+   */
+  @Override
+  public void delete(String path) throws IDicomWebClient.DicomWebException {
+    try {
+      HttpRequest httpRequest =
+          requestFactory.buildDeleteRequest(
+              new GenericUrl(serviceUrlPrefix + "/" + StringUtil.trim(path)));
+      httpRequest.execute();
+    } catch (HttpResponseException e) {
+      throw new DicomWebException(
+          String.format("delete: %d, %s", e.getStatusCode(), e.getStatusMessage()),
+          e,
+          e.getStatusCode(),
+          Status.ProcessingFailure);
+    } catch (IOException | IllegalArgumentException e) {
+      throw new IDicomWebClient.DicomWebException(e);
+    }
+  }
+
+  /**
+   * Reads the resource to figure out the study/series/instance UID path, deletes it and returns
+   * void.
+   *
+   * @param stream The unread stream containing the DICOM instance.
+   */
+  @Override
+  public void delete(InputStream stream) throws IDicomWebClient.DicomWebException {
+    try {
+      // TODO: HTTP 409 content has RetrieveURL as StudyUID, so we need to construct the instance
+      // path. Can we receive the RetrieveURL as the full instance path, as we're constructing here?
```

Comment:
Sadly this is not possible, because as you mentioned we only populate Study UID in the URL.